XpressAI/xpipes
# Use `@xai_component` to specify which components should be exposed in UI

# Description

This adds the ADR on using a decorator to specify what components should be available, and implements those changes.

## Pull Request Type


## Type of Change


# Tests

1. Components List
    1. Start Jupyter Lab
    2. Switch to component list
    3. Assert that all expected components are available

2. Specifying Component Colors (precondition: Jupyter Lab already running)
   1. Switch to component list
   2. Add / Change `color=` keyword on `@xai_component` decorator for the chosen component
   3. Click reload to see border color change of component


## Tested on?
